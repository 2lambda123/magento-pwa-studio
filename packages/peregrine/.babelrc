{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
