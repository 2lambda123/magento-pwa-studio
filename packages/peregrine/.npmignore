.eslintrc.js
.storybook
/lib/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
