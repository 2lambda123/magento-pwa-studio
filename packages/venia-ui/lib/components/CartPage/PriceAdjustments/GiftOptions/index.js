export { default, giftOptionsResolvers } from './giftOptions';
