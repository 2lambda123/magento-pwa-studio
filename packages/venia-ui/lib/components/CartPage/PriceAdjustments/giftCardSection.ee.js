import React from 'react';

import { Section } from '../../Accordion';
import GiftCards from '../GiftCards';

const GiftCardSection = props => {
    const { setIsCartUpdating } = props;
    return (
        <Section id={'gift_card'} title={'Apply Gift Card'}>
            <GiftCards setIsCartUpdating={setIsCartUpdating} />
        </Section>
    );
};

export default GiftCardSection;
