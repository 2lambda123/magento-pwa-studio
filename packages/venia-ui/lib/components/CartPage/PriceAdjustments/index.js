export { default } from './priceAdjustments';

export { giftOptionsResolvers } from './GiftOptions';
