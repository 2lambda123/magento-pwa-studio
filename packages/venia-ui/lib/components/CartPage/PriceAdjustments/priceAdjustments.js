import React from 'react';

import { Accordion, Section } from '../../Accordion';

import { mergeClasses } from '../../../classify';
import defaultClasses from './priceAdjustments.css';

const PriceAdjustments = props => {
    const classes = mergeClasses(defaultClasses, props.classes);

    return (
        <div className={classes.root}>
            <Accordion canOpenMultiple={true}>
                <Section
                    id={'shipping_method'}
                    isOpen={true}
                    title={'Select Shipping Method'}
                >
                    <a href="https://jira.corp.magento.com/browse/PWA-239">
                        Shipping Methods to be completed by PWA-239.
                    </a>
                </Section>
                <Section id={'coupon_code'} title={'Enter Coupon Code'}>
                    <a href="https://jira.corp.magento.com/browse/PWA-75">
                        Coupon Codes to be completed by PWA-75.
                    </a>
                </Section>
                <Section id={'gift_card'} title={'Apply Gift Card'}>
                    <a href="https://jira.corp.magento.com/browse/PWA-78">
                        Gift Cards to be completed by PWA-78.
                    </a>
                </Section>
                <Section id={'gift_options'} title={'See Gift Options'}>
                    <a href="https://jira.corp.magento.com/browse/PWA-178">
                        Gift Options to be completed by PWA-178.
                    </a>
                </Section>
            </Accordion>
        </div>
    );
};

export default PriceAdjustments;
