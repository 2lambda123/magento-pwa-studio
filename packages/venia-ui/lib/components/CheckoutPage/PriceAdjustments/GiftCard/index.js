import React from 'react';

export default () => <div>Will be handled in PWA-248</div>;
