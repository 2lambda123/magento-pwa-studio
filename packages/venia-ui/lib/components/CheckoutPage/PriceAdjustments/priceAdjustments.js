import React from 'react';
import { func } from 'prop-types';

import { mergeClasses } from '../../../classify';
import { Accordion, Section } from '../../Accordion';
import CouponCode from '../../CartPage/PriceAdjustments/CouponCode';
import GiftCardSection from '../../CartPage/PriceAdjustments/giftCardSection';
import GiftOptions from '../../CartPage/PriceAdjustments/GiftOptions';

import defaultClasses from './priceAdjustments.css';

/**
 * PriceAdjustments component for the Checkout page.

 * @param {Function} props.setPageIsUpdating callback that sets checkout page updating state
 */
const PriceAdjustments = props => {
    const classes = mergeClasses(defaultClasses, props.classes);

    const { setPageIsUpdating } = props;

    return (
        <div className={classes.root}>
            <Accordion canOpenMultiple={true}>
                <Section id={'coupon_code'} title={'Enter Coupon Code'}>
                    <CouponCode setIsCartUpdating={setPageIsUpdating} />
                </Section>
                <GiftCardSection setIsCartUpdating={setPageIsUpdating} />
                <Section id={'gift_options'} title={'See Gift Options'}>
                    <GiftOptions />
                </Section>
            </Accordion>
        </div>
    );
};

export default PriceAdjustments;

PriceAdjustments.propTypes = {
    setPageIsUpdating: func
};
