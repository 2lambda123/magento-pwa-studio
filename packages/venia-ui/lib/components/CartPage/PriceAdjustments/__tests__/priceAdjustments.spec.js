import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import PriceAdjustments from '../priceAdjustments';

test('it renders Venia price adjustments', () => {
    // Act.
    const instance = createTestInstance(<PriceAdjustments />);

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});
