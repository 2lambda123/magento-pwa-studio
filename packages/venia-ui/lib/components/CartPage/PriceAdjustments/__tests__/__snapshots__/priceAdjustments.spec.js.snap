// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Venia price adjustments 1`] = `
<div>
  <div>
    <div>
      <button
        onClick={[Function]}
      >
        <div>
          Select Shipping Method
        </div>
        <span>
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="18 15 12 9 6 15"
            />
          </svg>
        </span>
      </button>
      <div>
        <a
          href="https://jira.corp.magento.com/browse/PWA-239"
        >
          Shipping Methods to be completed by PWA-239.
        </a>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <div>
          Enter Coupon Code
        </div>
        <span>
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </span>
      </button>
      <div />
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <div>
          Apply Gift Card
        </div>
        <span>
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </span>
      </button>
      <div />
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <div>
          See Gift Options
        </div>
        <span>
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </span>
      </button>
      <div />
    </div>
  </div>
</div>
`;
